From 50042dc71b569501442145d89a0eac36852b64a5 Mon Sep 17 00:00:00 2001
From: Wincent Colaiuta <win@wincent.com>
Date: Wed, 28 May 2008 17:06:32 +0200
Subject: [PATCH 3/3] Prepend frozen gems to load path rather than appending

If a user freezes version "X" of a gem into their vendor directory
but a version "Y" of the same gem already exists on the system then we
were incorrectly loading version "Y". The fix is to prepend the frozen
gem's path rather than appending it to ensure that it gets loaded
instead.

Signed-off-by: Wincent Colaiuta <win@wincent.com>
---
 railties/lib/rails/gem_dependency.rb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/railties/lib/rails/gem_dependency.rb b/railties/lib/rails/gem_dependency.rb
index 703e282..97bd516 100644
--- a/railties/lib/rails/gem_dependency.rb
+++ b/railties/lib/rails/gem_dependency.rb
@@ -31,9 +31,9 @@ module Rails
         args << @requirement.to_s if @requirement
         gem *args
       else
-        $LOAD_PATH << File.join(unpacked_paths.first, 'lib')
+        $LOAD_PATH.unshift File.join(unpacked_paths.first, 'lib')
         ext = File.join(unpacked_paths.first, 'ext')
-        $LOAD_PATH << ext if File.exist?(ext)
+        $LOAD_PATH.unshift(ext) if File.exist?(ext)
         @frozen = true
       end
       @load_paths_added = true
-- 
1.5.5.3.dirty

